solver/llbsolver/bridge.go

```diff
@@ -202,26 +202,34 @@ type resultProxy struct {
 
 func newResultProxy(b *llbBridge, req frontend.SolveRequest) *resultProxy {
 	rp := &resultProxy{
-		def: req.Definition,
-	}
-	rp.cb = func(ctx context.Context) (solver.CachedResult, solver.BuildSources, error) {
-		res, bsrc, err := b.loadResult(ctx, req.Definition, req.CacheImports)
-		var ee *llberrdefs.ExecError
-		if errors.As(err, &ee) {
-			ee.EachRef(func(res solver.Result) error {
-				rp.errResults = append(rp.errResults, res)
-				return nil
-			})
-			// acquire ownership so ExecError finalizer doesn't attempt to release as well
-			ee.OwnerBorrowed = true
-		}
-		return res, bsrc, err
+		req: req,
+		b:   b,
 	}
 	return rp
 }
 
+func (rp *resultProxy) Clone() solver.ResultProxy {
+	rp.mu.Lock()
+	defer rp.mu.Unlock()
+
+	rp2 := &resultProxy{
```

Comment:
Why not just increase ref count for the existing object in here that gets decreased on `Release()`? Atm both clones would still trigger their individual solves (what is probably fine as solver knows how to duplicate but still why do it at all?).

I don't remember this `OwnerBorrowed` error attachements though that might need some double-checking when considering these cases.

Reply:
I think I wanted to avoid the case of being able to call `Release` on a single ref twice, to decrease the ref-count by two - ideally, there should be a way to detect that scenario, which involves returning a distinct value from the `Clone` function.

The other alternative I saw was to have a special `resultProxySplit` type similar to how the `SharedResult`s work, but I think it's roughly functionally equivalent to what's in the patch.